src/xtopcom/xdata/src/xdata_structures.cc

```diff
@@ -354,7 +354,7 @@ bool xreg_node_info::could_be_exchange() const noexcept {
 }
 
 bool xreg_node_info::could_be_fullnode() const noexcept {
-    return could_be<common::xnode_type_t::fullnode>(m_registered_miner_type);
+    return (could_be<common::xnode_type_t::fullnode>(m_registered_miner_type) && genesis());
```

Comment:
rename `could_be_fullnode()` to `could_be_fullnode_legacy()`
and add a new `could_be_fullnode()` which will call genesis() internally.

Reply:
no place to call this function .